src/internal/rest.rs

```diff
@@ -135,98 +399,156 @@ impl<'a> IronCoreRequest<'a> {
             Some(body),
             None,
             error_code,
-            auth.to_header(),
+            auth.to_auth_header(),
             move |server_resp| IronCoreRequest::deserialize_body(server_resp, error_code),
         )
     }
 
-    pub fn post_raw<B: DeserializeOwned>(
+    ///POST body to the resource at relative_url using IronCore authorization.
+    ///If the request fails a RequestError will be raised.
+    pub fn post<A: Serialize + 'a, B: DeserializeOwned + 'a>(
         &self,
         relative_url: &str,
-        body: &[u8],
+        body: &A,
         error_code: RequestErrorCode,
-        auth: &Authorization,
-    ) -> impl Future<Item = B, Error = IronOxideErr> {
-        let client = RClient::new();
-        let mut builder = client.request(
+        auth_b: crate::internal::auth_v2::AuthV2Builder<'a>,
+    ) -> impl Future<Item = B, Error = IronOxideErr> + 'a {
+        self.request_ironcore_auth::<A, _, _>(
+            relative_url,
             Method::POST,
-            format!("{}{}", self.base_url, relative_url).as_str(),
-        );
-
-        //We want to add the body as raw bytes
-        builder = builder.body(body.to_vec());
-
-        let req = builder
-            .headers(RAW_BYTES_HEADERS.clone())
-            .headers(auth.to_header());
-        IronCoreRequest::send_req(req, error_code.clone(), move |server_resp| {
-            IronCoreRequest::deserialize_body(server_resp, error_code.clone())
-        })
+            Some(body),
+            None,
+            error_code,
+            auth_b,
+            move |server_resp| IronCoreRequest::deserialize_body(server_resp, error_code),
+        )
+    }
+    pub fn post_raw<B: DeserializeOwned + 'a>(
```

Comment:
Isn't a lot of this code the same as the code in `request_ironcore_auth`? Maybe things could be refactored to share it.

Reply:
Yep, there is some duplication. I will take another look at it and consider what makes sense. A lot of this code is going to be refactored when we move to async/await too.

If `request_ironcore_auth` took the body as bytes instead of as `Serialize` that would solve it, but then all the json serialization would be duplicated or moved into a function of it's own. There are options, but I'm not sure it's worth the effort. I'll take another look. :)

Reply:
If you anticipate a big refactor when `async/await` arrives, it's probably fine to leave it for now. I was just surprised that `post` was so short, but `post_raw` was so long.